mpitests/test_adjoint_jacobian.py

```diff
@@ -62,265 +61,246 @@ def fixture_dev(request):
             qml.QuantumFunctionError,
-            match=message,
+            match="Adjoint differentiation method does not support measurement StateMP.",
         ):
-            dev.adjoint_jacobian(tape)
+            _ = dev.compute_derivatives(qs, config)
 
-    def test_finite_shots_warns(self):
+    @pytest.mark.parametrize("batch_obs", [True, False])
+    def test_finite_shots_warns(self, dev, batch_obs):
         """Tests warning raised when finite shots specified"""
 
-        dev = qml.device(device_name, wires=8, mpi=True, shots=1)
-
-        with qml.tape.QuantumTape() as tape:
-            qml.expval(qml.PauliZ(0))
+        qs = QuantumScript(
+            [qml.RX(1.23, 0)], [qml.expval(qml.Z(0))], shots=10, trainable_params=[0]
+        )
+        config = ExecutionConfig(gradient_method="adjoint", device_options={"batch_obs": batch_obs})
 
-        with pytest.warns(
-            UserWarning,
+        with pytest.raises(
+            qml.QuantumFunctionError,
             match="Requested adjoint differentiation to be computed with finite shots.",
         ):
-            dev.adjoint_jacobian(tape)
+            _ = dev.compute_derivatives(qs, config)
 
     def test_empty_measurements(self, dev):
         """Tests if an empty array is returned when the measurements of the tape is empty."""
 
-        with qml.tape.QuantumTape() as tape:
+        # with qml.tape.QuantumTape() as tape:
+        #     qml.RX(0.4, wires=[0])
+
+        def circuit():
             qml.RX(0.4, wires=[0])
+            return qml.expval(qml.PauliZ(0))
+
+        qnode = QNode(circuit, dev, diff_method="adjoint")
+
+        jac = qml.grad(qnode)()
 
-        jac = dev.adjoint_jacobian(tape)
         assert len(jac) == 0
 
-    def test_unsupported_op(self, dev):
+    @pytest.mark.parametrize("batch_obs", [True, False])
+    def test_unsupported_op(self, batch_obs, dev):
         """Test if a QuantumFunctionError is raised for an unsupported operation, i.e.,
         multi-parameter operations that are not qml.Rot"""
 
-        with qml.tape.QuantumTape() as tape:
-            qml.CRot(0.1, 0.2, 0.3, wires=[0, 1])
-            qml.expval(qml.PauliZ(0))
+        # with qml.tape.QuantumTape() as tape:
+        qs = QuantumScript(
+            [qml.CRot(0.1, 0.2, 0.3, wires=[0, 1])],
+            [qml.expval(qml.PauliZ(0))],
+            trainable_params=[0],
+        )
+        config = ExecutionConfig(gradient_method="adjoint", device_options={"batch_obs": batch_obs})
 
         with pytest.raises(
-            qml.QuantumFunctionError,
-            match="The CRot operation is not supported using the",
+            LightningException,
+            match="The operation is not supported using the adjoint differentiation method",
         ):
-            dev.adjoint_jacobian(tape)
+            _ = dev.compute_derivatives(qs, config)
 
-    def test_proj_unsupported(self, dev):
+    @pytest.mark.skip("WIP: Need a deep review if LGPU accept Projector")
```

Comment:
It is not clear to me if GPU with MPI has support for `Projector`

Reply:
We currently support this on LGPU via op diagonalization in Python. We maybe able to adapt the same logic for MPI too :thinking:  You don't need to add it to this PR though; just add the ticket to SC before resolving this conversation.

Reply:
Done!  [sc-74357]